### PR TITLE
Improve label handling in MasterListComponent

### LINKED ISSUE #1779: Handle primitive labeling in angular-material MasterListComponent
### Is your feature request related to a problem? Please describe.

When using a `ListWithDetail` on an array of strings, the label stays undefined even if the `options.labelRef` is set to `#`.

```json
{
  "type": "ListWithDetail",
  "scope": "#/properties/an-array-of-strings",
  "options": {
    "labelRef": "#"
  }
}
```

### Describe the solution you'd like

I'd like the label to be the value of the item. This could be achieved by modifiying the following in [master.ts](https://github.com/eclipsesource/jsonforms/blob/f40e95b5d6815b94f30460fa68e03b8e815a36be/packages/angular-material/src/other/master-detail/master.ts#L186)

```typescript
const dataIsPrimitive = d && typeof d !== 'object';
const masterItem = {
  label: get(d, labelRefInstancePath, dataIsPrimitive ? d : ''),
  data: d,
  path: `${path}.${index}`,
  schema,
  uischema: detailUISchema
};
```

### Describe alternatives you've considered

Using "my own" version of `MasterListComponent`.

### Framework

Angular

### RendererSet

Material

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by eneufeld**
The MasterListComponent did only show labels for object lists.
Furthermore the options property in the uischema was mandatory.
Also the property to use as label for objects had to be provided.

The fix now handles primitives, a missing options property and
in the case of a missing `labelRef` the first primitive property
is used. If no primitive property is available the first property in
the schema is used.

Fix #1779

### CODE REVIEW

File: packages/angular-material/src/other/master-detail/master.ts [hunk 1]
```diff
@@ -247,6 +249,15 @@ export class MasterListComponent extends JsonFormsArrayControl {
...
 }
 
+const getFirstUsableProperty = (schema: JsonSchema): string => {
+  const prop = Object.entries(schema?.properties).find(e => {
+    if(e[1].type === 'object' || e[1].properties) return false;
+    if(e[1].type === 'array' || e[1].items) return false;
+    return true;
+  });
+  return prop ? prop[0] : Object.keys(schema?.properties)[0];
+}
```
Comment (sdirix) [inline packages/angular-material/src/other/master-detail/master.ts:259]: There is a utility for this in `@jsonforms/core` which you can import: `getFirstPrimitiveProp`.

File: packages/angular-material/src/other/master-detail/master.ts [hunk 2]
```diff
@@ -170,7 +171,7 @@ export class MasterListComponent extends JsonFormsArrayControl {
     const controlElement = uischema as ControlElement;
     this.propsPath = props.path;
     const detailUISchema =
-      controlElement.options.detail ||
+      controlElement.options?.detail ||
       findUISchema(
```
Comment (sdirix) [inline packages/angular-material/src/other/master-detail/master.ts:175]: Not directly related to this PR but I just noticed this issue here: `findUISchema` already checks the `options.detail`, so we don't need to do this manually. Especially as our API also allows the strings `DEFAULT`, `GENERATED` and `REGISTERED` as valid `detail` values ;)

We can just hand over the `controlElement` as the next parameter after `'VerticalLayout'`, i.e.

```ts
const detailUISchema = findUISchema(
  props.uischemas,
  schema,
  `${controlElement.scope}/items`,
  props.path,
  'VerticalLayout',
  controlElement,
);
```

**Review by sdirix — CHANGES_REQUESTED** [review]
Thanks for the fix and the additional improvements! Noticed some minor issues.

File: packages/angular-material/src/other/master-detail/master.ts [hunk 3]
```diff
@@ -169,22 +170,22 @@ export class MasterListComponent extends JsonFormsArrayControl {
...
+        'VerticalLayout',
+        controlElement
       );
 
     const masterItems = (data || []).map((d: any, index: number) => {
-      const labelRefInstancePath = removeSchemaKeywords(
+      const labelRefInstancePath = controlElement.options?.labelRef && removeSchemaKeywords(
         controlElement.options.labelRef
       );
+      const isPrimitive = d && typeof d !== 'object';
```
Comment (sdirix) [inline packages/angular-material/src/other/master-detail/master.ts:186]: this is false for `0` and empty strings `''`

Comment (eneufeld) [reply packages/angular-material/src/other/master-detail/master.ts:186]: Will fix, JS and auto conversion of types ...

File: packages/examples/src/1779.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+};
+const uischema_number = {
+    "type": "ListWithDetail",
+    "scope": "#/properties/an-array-of-numbers"
+};
+
+registerExamples([
+    {
+        name: '1779',
+        label: 'List With Detail primitive (string)',
```
Comment (sdirix) [inline packages/examples/src/1779.ts:78]:
```suggestion
        label: 'List With Detail primitive (number)',
```

**Review by sdirix — APPROVED** [review]

**Pull request merged**